openapi/workflow_execution_service.openapi.yaml

```diff
@@ -788,14 +867,67 @@ components:
         task_logs:
           type: array
           items:
-            $ref: '#/components/schemas/Log'
+            $ref: '#/components/schemas/TaskLog'
           description: The logs, and other key info like timing and exit code, for each step in the workflow run.
         next_page_token:
           type: string
           description: >-
             A token which may be supplied as `page_token` in workflow run task list request to get the next page
             of results.  An empty string indicates there are no more items to return.
       description: The service will return a TaskListResponse when receiving a successful TaskListRequest.
+    TaskLog:
+      title: TaskLog
+      type: object
+      properties:
+        id:
+          type: string
+          description: A unique identifier which may be used to reference the task
```

Comment:
I have a feeling that `id` should be apart from this schema, similar to `RunStatus`. It would make more sense to me for the new endpoint. If we already supply the idea to get the task info, why would we need it in the response? And if we get a list of tasks, it's also maybe clearer to return, for each task, its ID and its log. I feel that this is semantically more appropriate as well.

But maybe I'm missing something!

Incidentally: I have just seen that there is some really weird regression (at least I think it's regression, because I'm pretty sure it used to be different, but too lazy to check now) in `RunListResponse`. I think it did (and it should) return an array of `RunStatus` (which is a run identifier and a state, basically), _not_ `State`. What good is an array of just _states_, if they don't link to specific runs??

Anyway, apart from the fact that this appears to be degraded, I think that same pattern should be used for the task logs as well.

Reply:
Imo `TaskLog` is more akin to `RunRequest` then `Log` or `RunSummary`. In this case the `TaskLog` IS the entity according to REST semantics, and there is not really any other place to put the identifier.

In most REST systems, responses that are a resource, typically have all of the relevant information pertaining to that resource. This may include an ID, but it also may include things like a `self_link` or other identifiers. I think NOT having the ID here actualy reduces the fucntionality of this PR, since the whole ID is to give tasks a stable identifier across runs.  If the ID is not part of the listing operation, then where would it come from?